### PR TITLE
Add ability to provide setupFiles to jest from modular configuration

### PR DESCRIPTION
**Pull request by LukeSheard**

### CODE REVIEW

**Comment by threepointone** [comment]
Looks good, could you fix the breaking CI build? (I think putting `export {}` should pass it.) Will stamp after CI passes.

**Comment by threepointone** [comment]
This also needs a changeset, probably patch.

File: packages/tree-view-for-tests/src/__tests__/tree.test.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@ test('it can serialise a folder', () => {
     ├─ .npmignore #1rstiru
     ├─ CHANGELOG.md
     ├─ README.md #r0jsfd
+    ├─ build
```
Comment (threepointone) [inline packages/tree-view-for-tests/src/__tests__/tree.test.ts:15]: this shouldn't be committed

Comment (LukeSheard) [reply packages/tree-view-for-tests/src/__tests__/tree.test.ts:15]: Yeah - this is annoying I will see if I can ignore this dir so it doesn't keep happening.

Comment (threepointone) [reply packages/tree-view-for-tests/src/__tests__/tree.test.ts:15]: we technically shouldn't ever have to run a manual build locally on these packages, were you experimenting with something?

**Review by threepointone — APPROVED** [review]
tentatively stamping, but please fix the failing snapshot test.

**Comment by LukeSheard** [comment]
Fixed hash now and changeset done.

**Pull request merged**